### DISCUSSION TITLE
FeatureRequest: Add app_icon Field to Cask

### DISCUSSION THREAD

**Discussion by shemeshg** [Homebrew/discussions#5228] [Casks]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

 I propose the addition of an app_icon field to Cask files.
 Currently, apps that require an icon rely on various unreliable algorithms to fetch icons from URLs. Having a distinct and explicit app_icon field would enhance consistency and reliability.

For example Applit uses:
```swift
            appIconView(iconURL: URL(string: "https://github.com/App-Fair/appcasks/releases/download/cask-\(cask.id)/AppIcon.png")!,
                        faviconURL: URL(string: "https://icon.horse/icon/\(cask.homepageURL.host ?? "")")!)
```

So I suggest something like
```Ruby
cask "mybrowser" do
  version "1.0"
  sha256 "abcdef1234567890"

  url "https://example.com/mybrowser-1.0.dmg"
  appcast "https://example.com/mybrowser-appcast.xml"
  homepage "https://example.com/mybrowser"

  app "MyBrowser.app"

  # New field for app icon
  app_icon "https://example.com/mybrowser-icon.png"
end
```

### Relevant casks

any

**Comment by SMillerDev** [comment]
I don't understand. What would the benefit of Homebrew users be? Apps already have an icon right?

**Reply by shemeshg** [reply]
CLI users will not benefit from having the `app_icon` tag for Homebrew casks; it is intended for ecosystem developers.

Ecosystem developers, such as `Applit`, cannot retrieve the app (or artifact) icon for applications they haven’t installed. Instead, they rely on favicons or third-party databases for this purpose.

By implementing this approach, Homebrew ecosystem developers will be able to present an icon for applications that users might want to install.

**Reply by SMillerDev** [reply]
I don't really see how it would benefit the Homebrew project to maintain volatile data like this. It would take a lot of effort to collect it and it would change nothing for Homebrew users.
